Update Time Manager

apcraig opened the pull request:
## PR checklist
- [X] Short (1 sentence) summary of your PR: 
    Update the time manager
- [X] Developer(s): 
    apcraig
- [X] Suggest PR reviewers from list in the column to the right.
- [X] Please copy the PR test results link or provide a summary of testing completed below.
    Full test suite run on cheyenne, all tests are bit-for-bit.  https://github.com/CICE-Consortium/Test-Results/wiki/cice_by_hash_forks#d17b86d4cd62e6fec54fb94d8048e7a9dc01feb0.  The test results suggest that it's not bit-for-bit, but because restart files are slightly different (but backwards compatible), the restart comparison failed.  A backup compare of the log files was added and those results indicate everything is bit-for-bit based on the log files for all cases.
  This has also been ported into RASM where the model is running and restarting exactly in coupled tests.
- How much do the PR code changes differ from the unmodified code? 
    - [X] bit for bit
    - [ ] different at roundoff level
    - [ ] more substantial 
- Does this PR create or have dependencies on Icepack or any other models?
    - [ ] Yes
    - [X] No
- Does this PR add any new test cases?
    - [ ] Yes
    - [X] No
- Is the documentation being updated? ("Documentation" includes information on the wiki or in the .rst files from doc/source/, which are used to create the online technical docs at https://readthedocs.org/projects/cice-consortium-cice/. A test build of the technical docs will be performed as part of the PR testing.)
    - [X] Yes
    - [ ] No, does the documentation need to be updated at a later time?
        - [ ] Yes
        - [ ] No 
- [X] Please provide any additional information or relevant details below:

This code has been rebased regularly to the current master.  It includes the following changes

- Refactor time manager
   - myear, mmonth, mday, msec are the new time manager prognostic variables
   - time is renamed to timesec and is a diagnostic, not prognostic variable
   - remove time_forc, was not needed or being used anyway
   - refactor ice_calendar, use calendar_type to define calendar after initialization
   - add several new subroutines to ice_calendar including
      - calc_timesteps   ! compute number of timesteps from npt_unit and npt
      - advance_timestep  ! advance model 1 timestep and update calendar                
      - calendar ! update model internal calendar/time information                      
      - set_date_from_timesecs ! set model date from time in seconds, needed for binary restarts backwards compatibility
      - compute_elapsed_days ! compute elapsed days since 0000-01-01                    
      - compute_days_between ! compute elapsed days between two dates                   
      - update_date          ! input date and delta date, compute new date              
      - calendar_date2time   ! convert date to time
      - calendar_time2date   ! convert time to date
      - compute_calendar_data ! compute info about calendar for a given year            
   - add month_init, day_init, sec_init to namelist
- Add various methods in ice_calendar to support adding and subtracting time, computing timesteps from other ways to specify time length, converting timesecs to nyr/month/day/sec and vv.
- Add npt_unit to go along with npt.  npt_unit can be 'y', 'm', 'd', 'h', 's', '1'.  '1' is the default and is timesteps for backwards compatibility.
- Add unit test capability
   - Add chkcal and helloworld tests
   - Modify scripts and build to support unit testing
   - Document unit testing capability
- Modify run lengths to leverage new "d" option in npt_unit, move away from timesteps
- Update JRA55_data implementation to improve performance relative to current master by reusing data that has already been read when possible.
- Add forcing timer to mpi and serial code
- Update comparelog script and call it automatically if restart files do not match
- Update documentation
- Update drivers/mct/cesm1/CICE_InitMod.F90 and CICE_RunMod.F90
   - use advance_timestep
   - get rid of use of time variable, shift to nyr, month, mday, sec
   - update calendar call
   - update usage of year_init
- Add section in user guide about coupling with new time manager
- Fix stuff unrelated to time manager
   - Fix bug in wght_file diagnostic in ice_domain.F90, was written before defined
   - Fix bug in bathymetry settings when max kmt is greater than hardcoded 40 levels
   - Add a few missing namelist variables recently added to documentation

apcraig commented:
@eclare108213 @dabail10 @phil-blain @rallard77 This is a DRAFT at this point, but please start reviewing if you can.

Still need to do a little code cleanup and we're going to run a multi-year RASM run to make sure it's reasonable.

This should address or partly address #558 (was fixed already in earlier version), #531, #343, #162.

apcraig commented:
See also #567

eclare108213 commented on cicecore/shared/ice_calendar.F90:
```diff
@@ -25,103 +27,132 @@ module ice_calendar
+         ice_calendar_360day    = '360day'        ! 360 day calendar with 30 days per month
 
-      real (kind=dbl_kind), parameter :: &
-        days_per_4c = 146097.0_dbl_kind, &
```
I see it in compute_calendar_data

apcraig replied:
I believe it's only in the icepack driver and is defined there.

eclare108213 replied:
My question still stands, whether the new time manager captures leap centuries correctly.  E.g. from https://www.infoplease.com/calendars/months-seasons/leap-year-explained:
> A leap year occurs every four years to help synchronize the calendar year with the solar year, or the length of time it takes the earth to complete its orbit aroundthe sun, which is about 365¼ days. The length of the solar year, however, is slightly less than 365¼ days-by about 11 minutes. To compensate for this discrepancy, the leap year is omitted three times every four hundred years. In other words, a century year cannot be a leap year unless it is divisible by 400. Thus 1700, 1800, and 1900 were not leap years, but 1600, 2000, and 2400 are leap years.

eclare108213 reviewed (commented):
Looks good so far.  A few comments/suggestions...

apcraig commented on cicecore/cicedynB/analysis/ice_history_shared.F90:
```diff
@@ -667,7 +667,7 @@ subroutine construct_filename(ncfile,suffix,ns)
       character(len=*), parameter :: subname = '(construct_filename)'
 
-        iyear = nyr + year_init - 1 ! set year_init=1 in ice_in to get iyear=nyr
+        iyear = nyr
```
nyr is now the model year.  There is no longer this dependence on year_init.  year_init just defines the model start year and plays no other role.  I think this was one of the main things that needed to change to make it easier to use and more robust.  I preserved use of the variables, nyr, month, mday, sec.  We could change "nyr" to "year" and "mday" to "day" or something else if it makes things clearer.

eclare108213 replied:
Since nyr has always (to me) meant 'number of the year' (i.e. a counter, POP convention again), could we use just 'year'?  Or is that used for something else?  iyear signalled the value of year as an integer quantity (although it might have been a real-values variable - I don't remember that kind of detail), e.g. 1996.  mday meant day of the month - I think that one's okay.

apcraig commented on cicecore/cicedynB/infrastructure/io/io_binary/ice_restart.F90:
```diff
@@ -31,6 +31,8 @@ module ice_restart
       public :: init_restart_write, init_restart_read, &
                 read_restart_field, write_restart_field, final_restart
 
+      real(kind=dbl_kind) :: time_forc = -99.   ! historic now local
```
Yes, it should be completely gone.  A grep indicated it was not used for anything.

eclare108213 replied:
Looks like time_forc is still here in the code (io_binary/ice_restart.F90)

apcraig commented on configuration/scripts/cice_decomp.csh:
```diff
@@ -98,6 +98,11 @@ else if (${grid} == 'tx1') then
     set blckx = 10; set blcky = 10
   endif
 
+else if (${grid} == 'none') then
```
It exists for unit testing.  The model may not run without a grid, but the unit tests may not need a grid, so I added this capability in the scripts.

eclare108213 replied:
please add a comment that says it's for unit testing, so it's not confused as a column configuration

apcraig commented on doc/source/cice_index.rst:
```diff
@@ -139,6 +139,7 @@ either Celsius or Kelvin units).
    "daymo", "number of days in one month", ""
    "daycal", "day number at end of month", ""
    "days_per_year", ":math:`\bullet` number of days in one year", "365"
+   "day_init", ":math:`\bullet` the initial day of the month", ""
```
The model can start on any year, month, day, and sec now.  I agree that since we don't have history restart capability, this could be an issue.  But that has always been the case.  CESM starts and stops on various days at times and the monthly averages are useless.  What we really need are history restarts.

eclare108213 replied:
Hopefully we have a note about that limitation in the documentation!

apcraig commented on cicecore/shared/ice_calendar.F90:
```diff
@@ -134,96 +165,169 @@ subroutine init_calendar
+      character(len=*),parameter :: subname='(advance_timestep)'
+
+      if (trim(npt_unit) /= '1') then
+         write(nu_diag,*) trim(subname),' ERROR npt_unit should be coverted to timesteps by now ',trim(npt_unit)
```
I fixed this.  With regard to npt and npt_unit, I kept the old npt variable name and just added npt_unit with the default being "1".  In that sense, it's fully backwards compatible.  But now we can define the run length in days, months, or other units which is something I think people wanted.  We could change the name of npt/npt_unit to "run_length_n" and "run_length_unit" or something like it, but then that adds a little extra work in adopting the new implementation.  My idea was to make it as backwards compatible as I could, but I'm fully open to the idea of using new variable and namelist names if they represent modified/new features.

apcraig commented on cicecore/shared/ice_calendar.F90:
```diff
@@ -134,96 +165,169 @@ subroutine init_calendar
       integer (kind=int_kind) :: &
          ns                         , & ! loop index
-         nyrp,mdayp,hourp           , & ! previous year, day, hour
+         nyrp,monthp,mdayp,hourp    , & ! previous year, day, hour
```
fixed.

apcraig commented:
@robertosinski ran a 40 year simulation with RASM with this version of the CICE model and the main conclusion is that the climate is the same.  I attach a number of figures including differences and season thickness plots.  I think it would be fine to use some of these plots in the release documentation if we like.

CICE61 is our current default CICE version and CICE62 is the version associated with this pull request.  It's really "pre 6.2".

Plots of September and March ice volume:

![CICE61_CICE62_RASM_G_ice_vol_SEP](https://user-images.githubusercontent.com/8182814/110817561-49593400-8241-11eb-8554-c49e13c2b833.png)
![CICE61_CICE62_RASM_G_ice_vol_MARCH](https://user-images.githubusercontent.com/8182814/110817566-49f1ca80-8241-11eb-9917-ed0cdd1aa709.png)

Plots of snow volume, ice volume, and ice extent:

![CICE61_CICE62_RASM_G_snow_vol](https://user-images.githubusercontent.com/8182814/110817572-4a8a6100-8241-11eb-80e9-5319db12b5c0.png)
![CICE61_CICE62_RASM_G_ice_vol](https://user-images.githubusercontent.com/8182814/110817573-4b22f780-8241-11eb-8b98-e35733958a94.png)
![CICE61_CICE62_RASM_G_ice_ext](https://user-images.githubusercontent.com/8182814/110817579-4bbb8e00-8241-11eb-8275-bb49161e9a11.png)

Plots of ice thickness for JFM, AMJ, JAS, OND, averaged 1980-2019:

![CICE6_2_OND_1980_2019_ice_thickness](https://user-images.githubusercontent.com/8182814/110817595-4d855180-8241-11eb-8e57-26b2c1ecf88d.png)
![CICE6_2_JFM_1980_2019_ice_thickness](https://user-images.githubusercontent.com/8182814/110817596-4d855180-8241-11eb-8b5f-6f6f76d23f37.png)
![CICE6_2_JAS_1980_2019_ice_thickness](https://user-images.githubusercontent.com/8182814/110817598-4e1de800-8241-11eb-9e2b-27b99dcfce62.png)
![CICE6_2_AMJ_1980_2019_ice_thickness](https://user-images.githubusercontent.com/8182814/110817600-4e1de800-8241-11eb-9e94-eaaa3818ca15.png)

Plots of ice thickness difference for JFM, AMJ, JAS, OND, 6.2 - 6.1, averaged 1980-2019:

![JFM_1980_2019_hi_diff_CICE62_minus_CICE61](https://user-images.githubusercontent.com/8182814/110817590-4cecbb00-8241-11eb-9336-14cf95a18940.png)
![AMJ_1980_2019_hi_diff_CICE62_minus_CICE61](https://user-images.githubusercontent.com/8182814/110817587-4c542480-8241-11eb-868b-4f8ee4ba8a4f.png)
![JAS_1980_2019_hi_diff_CICE62_minus_CICE61](https://user-images.githubusercontent.com/8182814/110817594-4cecbb00-8241-11eb-885a-8b865bf90f0c.png)
![OND_1980_2019_hi_diff_CICE62_minus_CICE61](https://user-images.githubusercontent.com/8182814/110817589-4c542480-8241-11eb-8878-bb3d24e27038.png)

apcraig commented:
This is ready for review and merge.

rallard77 commented:
Comparisons look great Tony!

dabail10 commented:
I'm going to try to run this in the CESM and standalone CICE. Sorry I haven't had time to look at this before now.

eclare108213 commented on doc/source/user_guide/ug_case_settings.rst:
```diff
@@ -355,6 +365,8 @@ dynamics_nml
    "", "", "", ""
    "``advection``", "``remap``", "linear remapping advection scheme", "``remap``"
    "", "``upwind``", "donor cell advection", ""
+   "``algo_nonlin``", "``anderson``", "use nonlinear anderson algorithm for implicit solver", "picard"
```
The algo_nonlin flag can be addressed with #575

apcraig replied:
@phil-blain?

phil-blain replied:
yes we could make it more specific. I thought that mentioning "implicit solver" in the description would have been enough,, but you are right that explicitely adding "VP", either in the description or the variable name itself would be best.

eclare108213 commented:
Nice comparison of 6.1 with this version.  @apcraig you said 6.1 is "our current default CICE version" but I think "our" refers to RASM, right?  Which version is it?  It would be interesting to run the QC comparison on a 5-year time span at the end of these runs to see whether it chokes on the changes.  Is that possible -- is the output in the right format?

eclare108213 reviewed (commented):
This PR is close.  Please respond to questions/suggestions in the conversation.  Thanks!

dabail10 reviewed (approved):
I don't really have any suggestions beyond what Elizabeth has said here. This is looking really great. I would really like to test this in the CESM, but I'm not quite ready for that. I guess I am a bit concerned about the answer changes in RASM. However, I guess it is not climate changing per se.

apcraig commented:
To follow up on the questions.

1. time_forc still exists in some form in the "binary" restart, because it was/is written to the restart even though it plays no role.  I left the name the same.  We cannot just remove it from the binary restart as this will create incompatibility issues.  We could change the name to "not_needed_variable" or something if that would be more desirable, but we still need a place holder for it in any case just in the binary restart read/write.
2. The current implementation supports 3 calendars.  365 day (no-leap using the standard variable days per month), 360 day (12 months of 30 days each), and what we're calling gregorian.  This is really a proleptic gregorian which has a leap day every 4 years but not every 100 years but every 400 years.  The current implementation does not need days_per_4c.  The calendar explicitly computes the days for each model year internally.  That is somewhat hardwired into the implementation now but only in a few places.   I believe the current implementation meets the gregorian requirement.  If we need to add other calendars, we can also do that fairly easily in the future.
3. The non bit-for-bit in the RASM tests is because we were comparing RASM with CICE6.2 (it's really pre-6.2) vs RASM with CICE6.0 (the 6.1 in the plots is a mislabeling, it's actually 6.0).  The ours means CICE in RASM and again, that's CICE6.0 not 6.1.  The non bit-for-bit of the multi decade RASM run likely has nothing to do with the calendar changes, but represents the changes in CICE between 6.0 and 6.2.   We could test a version of CICE pre-calendar with this version of CICE in RASM to see if it's bit-for-bit.  I expect it would be, but don't know for sure.  I'm not sure I feel that's necessary, but if others do, I could do it.
4. I just added some documentation about lack of history restarts
5. I added a comment to configuration/scripts/cice_decomp.csh to note the "none" grid is for unit testing.
6. With regard to internal variable names (nyr vs year vs iyear, etc).  What I did in the implementation is reuse some variables, add some variables, and delete some variables.  I didn't do a lot of renaming, but we could.  I think the biggest problem with variable names is that the prognostic variable names are (nyr, month, mday, sec).  These existed before and I was reluctant to change them just for the sake of change.  The namelist variable names to initialize the date are (year_init, month_init, day_init, sec_init).  year_init existed before and I added the other 3.  I would be in favor of changing the internal prognostic names from (nyr, month, mday, sec) to (year, month, day, sec).  There might be some conflicts in the current code, but I would then fix those.  If we want to do that, just let me know.  The other variable names that confuse me a bit are daymo and daycal, so I wouldn't mind having better names for those, but that's maybe less of a priority.

eclare108213 commented:
> 1. time_forc still exists in some form in the "binary" restart, because it was/is written to the restart even though it plays no role.  I left the name the same.  We cannot just remove it from the binary restart as this will create incompatibility issues.  We could change the name to "not_needed_variable" or something if that would be more desirable, but we still need a place holder for it in any case just in the binary restart read/write.

This makes sense.  Is it always written as -99 now?  If so, would it make sense to write the actual forcing date (or whatever time unit) at the time of the restart file dump?

> 2. The current implementation supports 3 calendars.  365 day (no-leap using the standard variable days per month), 360 day (12 months of 30 days each), and what we're calling gregorian.  This is really a proleptic gregorian which has a leap day every 4 years but not every 100 years but every 400 years.  The current implementation does not need days_per_4c.  The calendar explicitly computes the days for each model year internally.  That is somewhat hardwired into the implementation now but only in a few places.   I believe the current implementation meets the gregorian requirement.  If we need to add other calendars, we can also do that fairly easily in the future.

This is good, thanks.

> 3. The non bit-for-bit in the RASM tests is because we were comparing RASM with CICE6.2 (it's really pre-6.2) vs RASM with CICE6.0 (the 6.1 in the plots is a mislabeling, it's actually 6.0).  The ours means CICE in RASM and again, that's CICE6.0 not 6.1.  The non bit-for-bit of the multi decade RASM run likely has nothing to do with the calendar changes, but represents the changes in CICE between 6.0 and 6.2.   We could test a version of CICE pre-calendar with this version of CICE in RASM to see if it's bit-for-bit.  I expect it would be, but don't know for sure.  I'm not sure I feel that's necessary, but if others do, I could do it.

I think your test suite runs have established that it's BFB, right?  So if it's not in RASM, that's something to do with RASM, which could be tracked down and changed in the CICE repo later, if needed.  So I agree it's not necessary unless you want that sanity check for RASM purposes.

> 4. I just added some documentation about lack of history restarts
> 5. I added a comment to configuration/scripts/cice_decomp.csh to note the "none" grid is for unit testing.

thank you

> 6. With regard to internal variable names (nyr vs year vs iyear, etc).  What I did in the implementation is reuse some variables, add some variables, and delete some variables.  I didn't do a lot of renaming, but we could.  I think the biggest problem with variable names is that the prognostic variable names are (nyr, month, mday, sec).  These existed before and I was reluctant to change them just for the sake of change.  The namelist variable names to initialize the date are (year_init, month_init, day_init, sec_init).  year_init existed before and I added the other 3.  I would be in favor of changing the internal prognostic names from (nyr, month, mday, sec) to (year, month, day, sec).  There might be some conflicts in the current code, but I would then fix those.  If we want to do that, just let me know.  The other variable names that confuse me a bit are daymo and daycal, so I wouldn't mind having better names for those, but that's maybe less of a priority.

So the user interface has the nice names with '_init' suffixes.  I would prefer that the internal names match those, for clarity.  I'm not wed to keeping the older names.  daymo and daycal were useful for forcing, I think - they were day of the month (1, 2, ... 28 or 30 or 31) and day of the year (? 1, 2, ... 366).  Are they still used?

apcraig commented:
OK, I will update the internal prognostic names.  I think that's a good step.  

daymo and daycal are still very much used, mostly for forcing.  We really need to eventually switch to reading the time axis in the forcing file, but for now, we might as well stick with daymo and daycal, I guess.

as you note, time_forc is hardwired to -99 in the binary restart.  I'd almost prefer to not confuse the implementation and write something useful like the date of the last forcing call or something else.  It is not useful information, but by writing something that looks useful, it could lead to confusion.

The new calendar is bit-for-bit in the standalone model.  I may try to prove the same in RASM if it's not too difficult to do.  I'll need to port in a pre-new-calendar version CICE into RASM.  If it doesn't happen, I think it's OK too.

apcraig commented:
I have update the prognostic calendar names to myear, mmonth, mday, msec.  The "m" stands for model and provides some uniqueness in the source code to grep for them when needed.  I have run a full test suite on cheyenne with three compilers and all tests pass and are bit-for-bit.

I also was able to run some short tests in RASM comparing this version of the model and the version prior to this PR and the results are identical for 5 days in a gx1 (1 degree global) and an ar9 (9 km Arctic resolution) configuration.

phil-blain commented on cicecore/shared/ice_calendar.F90:
```diff
@@ -134,99 +164,172 @@ subroutine init_calendar
+      mday=day_init     ! day of the month
+      msec=sec_init     ! seconds into date
       istep1 = istep0   ! number of steps at current timestep
                         ! real (dumped) or imagined (use to set calendar)
```
likewise, this comment could be removed to avoid confusion
~~~fortran
                        ! real (dumped) or imagined (use to set calendar)
~~~

apcraig replied:
The istep0 question is a reasonable one.  I think I left it in there so a user could define the initial step number as something other than 0 if they wanted.  The key is that the steps still need to be counted and tracked on restart and they are.  I think if we remove istep0 from namelist and it is effectively hardwired to 0, then we can probably get rid of it in the entire code.  I think we should discuss further.

phil-blain reviewed (commented):
Hi @apcraig, sorry I did not have time to review before the merge. I took a look and left some comments. Thanks for this significant piece of work!

apcraig commented on doc/source/user_guide/ug_implementation.rst:
```diff
@@ -759,7 +821,8 @@ is now a character string corresponding to ``histfreq`` or ‘x’ for none.
 files, no matter what the frequency is.) If there are no namelist flags
 with a given ``histfreq`` value, or if an element of ``histfreq_n`` is 0, then
 no file will be written at that frequency. The output period can be
-discerned from the filenames.
+discerned from the filenames.  Because all history is average output, it's
+not possible to write instananeous output at any frequency except every timestep.
```
I don't have the answer to this question, but it's a good one.  I'm not super familiar with all the details of the history mechanism.  @dabail10.

dabail10 replied:
This is a good question. When I implemented the multiple stream capability years ago, I did not think about the hist_avg variable. I kind of just made the assumption that it was all or none for the averaging. So, if hist_avg = .false. all streams should be instantaneous. However, I think it would be nice to make this an array, so that we could pick and choose for each stream. There is also some issues in the code where it checks only the first stream for a value that is not 'x'. This means that the first one must always be defined or it turns off history completely.

phil-blain replied:
OK, so if I understand correctly, the doc as modified by this PR is wrong ?

dabail10 replied:
I haven't actually tested the code with hist_avg = .false.

daveh150 commented:
We run with hist_avg = .false. routinely. It seems to work as expected, i.e., we have

 

    histfreq       = 'd','x','x','x','x'

    histfreq_n     =  1 , 1 , 1 , 1 , 1

    hist_avg       = .false.

 

and it writes an instantaneous snapshot every day. Or at least I think it does? Am I misunderstanding this thread?

 

Thanks,

David

 

From: David A. Bailey ***@***.***> 
Sent: Thursday, April 1, 2021 9:53 AM
To: CICE-Consortium/CICE ***@***.***>
Cc: Subscribed ***@***.***>
Subject: Re: [CICE-Consortium/CICE] Update Time Manager (#566)

 

@dabail10 commented on this pull request.

  _____  

In doc/source/user_guide/ug_implementation.rst <https://github.com/CICE-Consortium/CICE/pull/566#discussion_r605722079> :

> -discerned from the filenames.
+discerned from the filenames.  Because all history is average output, it's
+not possible to write instananeous output at any frequency except every timestep.

I haven't actually tested the code with hist_avg = .false.

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub <https://github.com/CICE-Consortium/CICE/pull/566#discussion_r605722079> , or unsubscribe <https://github.com/notifications/unsubscribe-auth/AE52VPBRQ3EV22SCW5E7XC3TGSCGPANCNFSM4YJK26LA> .  <https://github.com/notifications/beacon/AE52VPFFWMS2GMZ4S3MVKZLTGSCGPA5CNFSM4YJK26LKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOEVKWT7Q.gif>

dabail10 commented:
This is great. Actually, I think for the Quality Control tests, we write instantaneous daily output. However, I think the question is if you have histfreq = 'm','d','x','x','x' say, with hist_avg = .false. does this mean instantaneous monthly and daily?

phil-blain commented:
To be clear,  my question here https://github.com/CICE-Consortium/CICE/pull/566#discussion_r605053734 was: is the change to the documentation introduced in this PR right or wrong. I think it is wrong: it now states:
> it's not possible to write instananeous output at any frequency except every timestep.

which is false; you *can* have instantaneous output at other frequency that `dt`.

daveh150 commented:
Thank you Philippe, I agree instantaneous outputs seem to work fine at specified frequencies other than every timestep. 

 

@dbailey, in your example specifying instantaneous output for month and day seems redundant, if you write the same information at each interval. Since there is no averaging, the instantaneous monthly output is probably the same as daily output.

 

Having said that, I could see a scenario where you only want specific output on a monthly basis instead of daily… example could be only writing the ice category information on a monthly basis, but the aggregate on daily basis. 

 

 

From: Philippe Blain ***@***.***> 
Sent: Thursday, April 1, 2021 10:14 AM
To: CICE-Consortium/CICE ***@***.***>
Cc: David Hebert, Code 7322 ***@***.***>; Comment ***@***.***>
Subject: Re: [CICE-Consortium/CICE] Update Time Manager (#566)

 

To be clear, my question here #566 (comment) <https://github.com/CICE-Consortium/CICE/pull/566#discussion_r605053734>  was: is the change to the documentation introduced in this PR right or wrong. I think it is wrong: it now states:

it's not possible to write instananeous output at any frequency except every timestep.

which is false; you can have instantaneous output at other frequency that dt.

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub <https://github.com/CICE-Consortium/CICE/pull/566#issuecomment-811976427> , or unsubscribe <https://github.com/notifications/unsubscribe-auth/AE52VPGSFT4XVDVVIH4DLKDTGSESTANCNFSM4YJK26LA> .  <https://github.com/notifications/beacon/AE52VPBXOA4U7YEJ47XWPZTTGSESTA5CNFSM4YJK26LKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOGBS4N2Y.gif>

apcraig commented:
I'll fix the documentation in PR #586 now.
